overlord/devicestate/devicestate_install_api_test.go

```diff
@@ -0,0 +1,439 @@
+	s.AddCleanup(restore)
+
+	s.state.Lock()
+	defer s.state.Unlock()
+
+	// Create change
+	label := "classic"
+	chg := s.state.NewChange("install-step-finish", "finish setup of run system")
+	finishTask := s.state.NewTask("install-finish", "install API finish step")
+	finishTask.Set("system-label", label)
+	finishTask.Set("on-volumes", ginfo.Volumes)
+
+	chg.AddTask(finishTask)
+
+	// now let the change run - some checks will happen in the mocked functions
+	s.state.Unlock()
+	defer s.state.Lock()
+
+	s.settle(c)
+
+	s.state.Lock()
+	defer s.state.Unlock()
+
+	// Checks now
+	c.Check(chg.Err(), ErrorMatches, `cannot perform the following tasks:
+- install API finish step \(cannot load assertions for label "classic": no seed assertions\)`)
+}
+
+func (s *deviceMgrInstallAPISuite) testInstallSetupStorageEncryption(c *C) {
+	// Mock label
+	label := "classic"
+	isClassic := true
+	gadgetSnapPath, kernelSnapPath, ginfo, mountCmd := s.mockLabel(c, label, isClassic)
+
+	// Simulate system with TPM
+	restore := devicestate.MockSecbootCheckTPMKeySealingSupported(func() error { return nil })
+	s.AddCleanup(restore)
+
+	// Mock encryption of partitions
+	encrytpPartCalls := 0
+	restore = devicestate.MockInstallEncryptPartitions(func(onVolumes map[string]*gadget.Volume, encryptionType secboot.EncryptionType, model *asserts.Model, gadgetRoot, kernelRoot string, perfTimings timings.Measurer) (*install.EncryptionSetupData, error) {
+		encrytpPartCalls++
+		return &install.EncryptionSetupData{}, nil
+	})
+	s.AddCleanup(restore)
+
+	s.state.Lock()
+	defer s.state.Unlock()
+
+	// Create change
+	chg := s.state.NewChange("install-step-setup-storage-encryption",
+		"Setup storage encryption")
+	encryptTask := s.state.NewTask("install-setup-storage-encryption",
+		"install API set-up encryption step")
+	encryptTask.Set("system-label", label)
+	encryptTask.Set("on-volumes", ginfo.Volumes)
+	chg.AddTask(encryptTask)
+
+	// now let the change run - some checks will happen in the mocked functions
+	s.state.Unlock()
+	defer s.state.Lock()
+
+	s.settle(c)
+
+	// Make sure that if anything was stored in cache it is removed after test is run
+	s.AddCleanup(func() {
+		s.state.Lock()
+		defer s.state.Unlock()
+		devicestate.CleanUpEncryptionSetupDataInCache(s.state, label)
+	})
+
+	s.state.Lock()
+	defer s.state.Unlock()
+
+	// Checks now
+	c.Check(chg.Err(), IsNil)
+	gadgetDir := filepath.Join(dirs.SnapRunDir, "snap-content/gadget")
+	kernelDir := filepath.Join(dirs.SnapRunDir, "snap-content/kernel")
+	c.Check(mountCmd.Calls(), DeepEquals, [][]string{
+		{"systemd-mount", gadgetSnapPath, gadgetDir},
+		{"systemd-mount", kernelSnapPath, kernelDir},
+		{"systemd-mount", "--umount", gadgetDir},
+		{"systemd-mount", "--umount", kernelDir},
+	})
+	c.Check(encrytpPartCalls, Equals, 1)
+	// Check that some data has been stored in the change
+	apiData := make(map[string]interface{})
+	c.Check(chg.Get("api-data", &apiData), IsNil)
+	_, ok := apiData["encrypted-devices"]
```

Comment:
should this check more about the api data?

Reply:
Unfortunately it is not possible to fill the partitions in EncryptionSetupData from these tests - that is tested from install tests though.